bin/check-elb-instances-endpoints.rb

```diff
@@ -0,0 +1,146 @@
+#! /usr/bin/env ruby
+#
+# check-elb-instances-endpoints
+#
+# DESCRIPTION:
+#   This plugin checks for a HTTP 200 against each instance attached to a ELB.
+#
+# OUTPUT:
+#   plain-text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: aws-sdk
+#   gem: sensu-plugin
+#
+# USAGE:
+#
+#   Check each instance on the elb returns a HTTP200 on the path /
+#   check-elb-instances-endpoints.rb -n ELBNAME -p /
+#
+#   Check each instance on the elb returns a HTTP200 on the path / and use https
+#   check-elb-instances-endpoints.rb -n ELBNAME -p / -s
+#
+#   Check each instance on the elb returns a HTTP200 on the path / using port 8080
+#   check-elb-instances-endpoints.rb -n ELBNAME -p / -P 8080
+#
+#   Check each instance on the elb returns a HTTP200 on the path / in a given AWS region
+#   check-elb-instances-endpoints.rb -n ELBNAME -p / -r eu-west-1
+#
+# LICENSE:
+#    MIT License
+#
+#    Copyright (c) 2017 Claranet
+#
+#    Permission is hereby granted, free of charge, to any person obtaining a copy
+#    of this software and associated documentation files (the "Software"), to deal
+#    in the Software without restriction, including without limitation the rights
+#    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#    copies of the Software, and to permit persons to whom the Software is
+#    furnished to do so, subject to the following conditions:
+#
+#    The above copyright notice and this permission notice shall be included in all
+#    copies or substantial portions of the Software.
+#
+#    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+#    SOFTWARE.
+#
+
+require 'sensu-plugin/check/cli'
+require 'aws-sdk-v1'
```

Comment:
with this being a new check we should write this against the v2 api. Eventually we will be going through and updating the checks for v2 and dropping v1 support due to version conflicts with newer versions of some gems (json if I recall correctly) with newer sensu setups.

Reply:
yes this is a really big issue which prevents us from upgrading to 1.x

Reply:
Can we re-write this to be against v2?

Reply:
:+1: